reconcile/utils/oc.py

```diff
@@ -690,57 +690,66 @@ def get_obj_root_owner(self, ns, obj, allow_not_found=False,
-        for c in containers:
-            for e in c.get('envFrom', []):
-                try:
-                    if e['secretRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-            for e in c.get('env', []):
-                try:
-                    if e['valueFrom']['secretKeyRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-        return False
+    def secret_used_in_pod(self, name, pod):
+        used_resources = self.get_resources_used_in_pod_spec(pod["spec"], "Secret")
+        return name in used_resources
+
+    def configmap_used_in_pod(self, name, pod):
+        used_resources = self.get_resources_used_in_pod_spec(pod["spec"], "ConfigMap")
+        return name in used_resources
 
     @staticmethod
-    def configmap_used_in_pod(name, pod):
-        volumes = pod['spec']['volumes']
-        for v in volumes:
-            volume_item = v.get('configMap', {})
-            try:
-                if volume_item['name'] == name:
-                    return True
-            except KeyError:
-                continue
-        containers = pod['spec']['containers']
-        for c in containers:
-            for e in c.get('envFrom', []):
-                try:
-                    if e['configMapRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-            for e in c.get('env', []):
+    def get_resources_used_in_pod_spec(spec: Dict[str, Any], kind: str) -> Dict[str, Set[str]]:
+        if kind not in ("Secret", "ConfigMap"):
+            raise KeyError(f"unsupported kind for get_resources_used_in_pod_spec: {kind}")
+        if kind == "Secret":
+            volume_kind = "secret"
+            volume_kind_ref = "secretName"
+            env_from_kind = "secretRef"
+            env_kind = "secretKeyRef"
+            env_ref = "name"
+        elif kind == "ConfigMap":
+            volume_kind = "configMap"
+            volume_kind_ref = "name"
+            env_from_kind = "configMapRef"
+            env_kind = "configMapKeyRef"
+            env_ref = "name"
+
+        resources = {}
+        volumes = spec.get("volumes")
+        if volumes:
+            for v in volumes:
                 try:
-                    if e['valueFrom']['configMapKeyRef']['name'] == name:
-                        return True
-                except KeyError:
+                    resource_name = v[volume_kind][volume_kind_ref]
+                    resources.setdefault(resource_name, set())
+                except (KeyError, TypeError):
                     continue
-        return False
+        containers = spec["containers"]
+        init_containers = spec.get("initContainers")
+        if init_containers:
+            containers += init_containers
+        for c in containers:
+            env_from = c.get("envFrom")
```

Comment:
Same as above, either of the options below saves you one `if`, which makes the code easier to read.
```python
            env_from = c.get("envFrom", [])
            env_from = c["envFrom"] or []
```